src/shogun/base/SGObject.h

```diff
@@ -1095,12 +1125,8 @@ template <typename T>
 CSGObject* get_if_possible(const CSGObject* obj, const std::string& name, GetByNameIndex how)
 {
 	CSGObject* result = nullptr;
-	try
-	{
-		// there is no "has" that checks for array types, so check implicitly
-		result = obj->get<T>(name, how.m_index);
-	}
-	catch (const std::exception&) {}
+	// there is no "has" that checks for array types, so check implicitly
+	result = obj->get<T>(name, how.m_index, std::nothrow);
```

Comment:
ah I see. Fine then!

Reply:
This change was done to throw correctly the exceptions when the `get(call, index)` does not work. The previous implementation was causing errors with the interfaces.

Reply:
ah good catch then!